core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -56,63 +77,111 @@
 @SdkPublicApi
 public final class ClientOverrideConfiguration
         implements ToCopyableBuilder<ClientOverrideConfiguration.Builder, ClientOverrideConfiguration> {
+    /**
+     * The set of options modified by this ClientOverrideConfiguration. This is used when the ClientOverrideConfiguration
+     * is created from a {@link SdkClientConfiguration} to filter out properties that this object doesn't use.
+     *
+     * This is important so that unrelated configuration values don't "pass through" from when this object is created
+     * from a SdkClientConfiguration and then converted back.
+     */
+    private static final Set<ClientOption<?>> CLIENT_OVERRIDE_OPTIONS;
+
+    /**
+     * The set of options that can be visible from this ClientOverrideConfiguration, but can't be modified directly. For
+     * example, when this ClientOverrideConfiguration is created from an SdkClientConfiguration, we want the
+     * {@link SdkClientOption#COMPRESSION_CONFIGURATION} to be visible to {@link #compressionConfiguration()} even though
+     * the setting that this object manipulates is {@link SdkClientOption#CONFIGURED_COMPRESSION_CONFIGURATION}.
+     *
```

Comment:
When it has both a derived version that the core uses, and a non-derived version that the configuration objects use.

This happens when it's an object that needs to have modifications applied after it's modified by a plugin. 

Here, the compression configuration has it because if someone sets an empty compression configuration, we want to add the default compression-enabled and compression-threshold settings. The scheduled executor service has it because it allows us to have executors we create be managed and ones the customer specifies be unmanaged.

Reply:
Okay I see, that makes sense!